protocol/message_handler.go

```diff
@@ -722,3 +731,32 @@ func (m *MessageHandler) HandleEmojiReaction(state *ReceivedMessageState, pbEmoj
 
 	return nil
 }
+
+func (m *MessageHandler) HandleGroupChatInvitation(state *ReceivedMessageState, pbGHInvitations protobuf.GroupChatInvitation) error {
+	logger := m.logger.With(zap.String("site", "HandleGroupChatInvitation"))
```

Comment:
Here you can check back if you have requests in the state so you don't process them multiple times, as messages are received multiple times.